external/avro/src/main/scala/org/apache/spark/sql/avro/SparkAvroDatumReader.scala

```diff
@@ -144,39 +144,39 @@ class SparkAvroDatumReader[T](
+    case (NULL, NullType) => makeNullReader(avroType)
 
     // TODO: we can avoid boxing if future version of avro provide primitive accessors.
-    case (BOOLEAN, BooleanType) => getBooleanReader(avroType)
-    case (INT, IntegerType) => getIntReader(avroType)
-    case (INT, DateType) => getIntAsDateReader(avroType)
-    case (LONG, LongType) => getLongReader(avroType)
+    case (BOOLEAN, BooleanType) => makeBooleanReader(avroType)
+    case (INT, IntegerType) => makeIntReader(avroType)
+    case (INT, DateType) => makeIntAsDateReader(avroType)
+    case (LONG, LongType) => makeLongReader(avroType)
     case (LONG, TimestampType) => avroType.getLogicalType match {
       // For backward compatibility, if the Avro type is Long and it is not logical type
       // (the `null` case), the value is processed as timestamp type with millisecond precision.
-      case null | _: TimestampMillis => getTimestampMillisReader(avroType)
-      case _: TimestampMicros => getTimestampMicrosReader(avroType)
+      case null | _: TimestampMillis => makeTimestampMillisReader(avroType)
+      case _: TimestampMicros => makeTimestampMicrosReader(avroType)
       case other => throw new IncompatibleSchemaException(
         s"Cannot convert Avro logical type ${other} to Catalyst Timestamp type.")
     }
 
     // Before we upgrade Avro to 1.8 for logical type support, spark-avro converts Long to Date.
     // For backward compatibility, we still keep this conversion.
-    case (LONG, DateType) => getLongAsDateReader(avroType)
-    case (FLOAT, FloatType) => getFloatReader(avroType)
-    case (DOUBLE, DoubleType) => getDoubleReader(avroType)
-    case (STRING, StringType) => getStringReader(avroType)
-    case (ENUM, StringType) => getEnumAsStringReader(avroType)
-    case (FIXED, BinaryType) => getFixedAsBinaryReader(avroType)
-    case (BYTES, BinaryType) => getBytesAsBinaryReader(avroType)
-    case (FIXED, d: DecimalType) => getFixedAsDecimalReader(avroType, d)
-    case (BYTES, d: DecimalType) => getBytesAsDecimalReader(avroType, d)
-    case (RECORD, st: StructType) => getRecordReader(avroType, st, path, reuseObj)
-    case (ARRAY, ArrayType(elementType, _)) => getArrayReader(avroType, elementType, path)
+    case (LONG, DateType) => makeLongAsDateReader(avroType)
+    case (FLOAT, FloatType) => makeFloatReader(avroType)
+    case (DOUBLE, DoubleType) => makeDoubleReader(avroType)
+    case (STRING, StringType) => makeStringReader(avroType)
+    case (ENUM, StringType) => makeEnumAsStringReader(avroType)
+    case (FIXED, BinaryType) => makeFixedAsBinaryReader(avroType)
+    case (BYTES, BinaryType) => makeBytesAsBinaryReader(avroType)
+    case (FIXED, d: DecimalType) => makeFixedAsDecimalReader(avroType, d)
+    case (BYTES, d: DecimalType) => makeBytesAsDecimalReader(avroType, d)
+    case (RECORD, st: StructType) => makeRecordReader(avroType, st, path, reuseObj)
+    case (ARRAY, ArrayType(elementType, _)) => makeArrayReader(avroType, elementType, path)
```

Comment:
Changing names of methods from getXYZReader to makeXYZReader since that better reflects the methods.